enterprise/frontend/src/metabase-enterprise/api/transform.ts

```diff
@@ -66,6 +66,25 @@ export const transformApi = EnterpriseApi.injectEndpoints({
       }),
       invalidatesTags: (_, error, id) =>
         invalidateTags(error, [idTag("transform", id), tag("table")]),
+      onQueryStarted: async (id, { dispatch, queryFulfilled }) => {
+        const patchResult = dispatch(
+          transformApi.util.updateQueryData("getTransform", id, (draft) => {
+            draft.last_run = {
+              id: -1,
+              status: "started",
+              start_time: new Date().toISOString(),
+              end_time: null,
+              message: null,
+              run_method: "manual",
+            };
+          }),
+        );
+        try {
+          await queryFulfilled;
```

Comment:
We can do even better here and also assign the right `id` to the `last_run`. See https://github.com/metabase/metabase/pull/62387/files#diff-adf3bbcaf485af8f00dc41c435c654109361881ccf4e1d440736b068630a6dc4R91-R97